src/deployment/deploy.py

```diff
@@ -594,6 +602,38 @@ def assign_scaleset_identity_role(self) -> None:
             OnefuzzAppRole.ManagedNode,
         )
 
+    def assign_user_access(self) -> None:
+        logger.info("assinging user access to service principal")
+        app = get_application(
+            display_name=self.application_name,
+            subscription_id=self.get_subscription_id(),
+        )
+        user = get_signed_in_user(self.subscription_id)
+
+        if app:
+            sp = get_service_principal(app["appId"], self.subscription_id)
+            # Update appRoleAssignmentRequired if necessary
+            if not sp["appRoleAssignmentRequired"]:
+                logger.warning(
+                    "The Service Principal does not have 'appRoleAssignmentRequired' set to True."
```

Comment:
i would simplify the whole warning like this:

> the service is not currently configured to require a role assignment to access it. 
This means that any authenticated user can access the service.
To change this behavior enble `Assignment required?` on the service principal in the azure portal

The problem is `appRoleAssignmentRequired` is the value you set in the msgraph rest api but in the portal it is called a `Assignment required`

Reply:
Done!